CondTools/Geometry/plugins/PPSGeometryBuilder.cc

```diff
@@ -21,244 +19,78 @@
 #include "FWCore/ServiceRegistry/interface/Service.h"
 #include "CondCore/DBOutputService/interface/PoolDBOutputService.h"
 
-#include "DetectorDescription/Core/interface/DDCompactView.h"
-#include "DetectorDescription/Core/interface/DDFilteredView.h"
-
-#include "DataFormats/CTPPSDetId/interface/TotemRPDetId.h"
-#include "DataFormats/CTPPSDetId/interface/TotemTimingDetId.h"
-#include "DataFormats/CTPPSDetId/interface/CTPPSPixelDetId.h"
-#include "DataFormats/CTPPSDetId/interface/CTPPSDiamondDetId.h"
-#include "Geometry/VeryForwardGeometryBuilder/interface/CTPPSDDDNames.h"
-
 #include "Geometry/Records/interface/IdealGeometryRecord.h"
 #include "Geometry/Records/interface/VeryForwardIdealGeometryRecord.h"
-
+#include "DetectorDescription/DDCMS/interface/DDCompactView.h"
 #include "CondFormats/GeometryObjects/interface/PDetGeomDesc.h"
-
-#include <regex>
-
-//----------------------------------------------------------------------------------------------------
+#include "Geometry/VeryForwardGeometryBuilder/interface/DetGeomDescBuilder.h"
 
 class PPSGeometryBuilder : public edm::one::EDAnalyzer<> {
 public:
   explicit PPSGeometryBuilder(const edm::ParameterSet&);
   void analyze(const edm::Event&, const edm::EventSetup&) override;
 
 private:
-  void buildPDetGeomDesc(DDFilteredView*, PDetGeomDesc*);
-  uint32_t getGeographicalID(DDFilteredView*);
+  void buildSerializableDataFromGeoInfo(PDetGeomDesc* serializableData, const DetGeomDesc* geoInfo, int& counter);
 
   std::string compactViewTag_;
   edm::ESWatcher<IdealGeometryRecord> watcherIdealGeometry_;
-  edm::Service<cond::service::PoolDBOutputService> dbservice_;
+  edm::Service<cond::service::PoolDBOutputService> dbService_;
 };
 
-//----------------------------------------------------------------------------------------------------
-
 PPSGeometryBuilder::PPSGeometryBuilder(const edm::ParameterSet& iConfig)
     : compactViewTag_(iConfig.getUntrackedParameter<std::string>("compactViewTag", "XMLIdealGeometryESSource_CTPPS")) {}
 
-//----------------------------------------------------------------------------------------------------
-
+/*
+ * Save PPS geo to DB.
+ */
 void PPSGeometryBuilder::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
-  edm::ESHandle<DDCompactView> cpv;
+  edm::ESHandle<cms::DDCompactView> myCompactView;
 
-  // Get DDCompactView from IdealGeometryRecord
   if (watcherIdealGeometry_.check(iSetup)) {
-    iSetup.get<IdealGeometryRecord>().get(compactViewTag_.c_str(), cpv);
+    edm::LogInfo("PPSGeometryBuilder") << "Got IdealGeometryRecord ";
+    iSetup.get<IdealGeometryRecord>().get(compactViewTag_.c_str(), myCompactView);
```

Comment:
This needs to be changed to use ESGetToken soon, see
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideHowToGetDataFromES#In_ED_module
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideHowToGetDataFromES#Getting_data_from_EventSetup_wit

(although we're almost done providing a simpler way to call `esConsumes()`)